pkg/reconciler/configuration/configuration.go

```diff
@@ -191,23 +194,32 @@ func (c *Reconciler) reconcile(ctx context.Context, config *v1alpha1.Configurati
 		config.Status.MarkLatestCreatedFailed(lcr.Name, rc.Message)
 		c.Recorder.Eventf(config, corev1.EventTypeWarning, "LatestCreatedFailed",
 			"Latest created revision %q has failed", lcr.Name)
-
+		if err = c.findAndSetLatestReadyRevision(config); err != nil {
+			return fmt.Errorf("failed to find and set latest ready revision: %w", err)
+		}
```

Comment:
184-188 look like a special case of this, so perhaps we can just eliminate that and sink this call?  I like getting as much mileage through code as possible, so if we can make this a single code path 🎉